From 8c62bd9deecbd7415ca67b73724eb0105716891a Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Sat, 9 Aug 2025 14:50:38 +0100
Subject: [PATCH 5/6] Make xpack settings default to FIPS

Ensure autoconfigure is disabled by default, as it tries to create 3DES PEM files which are unapproved and not in bcfks format.

Ensure password hashing algorithm always defaults to pbkdf2 stretch.
---
 .../org/elasticsearch/xpack/core/XPackSettings.java    | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java b/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java
index 81b56feb8ee..782f5ddd4c8 100644
--- a/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java
+++ b/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java
@@ -208,7 +208,7 @@ public class XPackSettings {
      */
     public static final Setting<Boolean> SECURITY_AUTOCONFIGURATION_ENABLED = Setting.boolSetting(
         "xpack.security.autoconfiguration.enabled",
-        true,
+        false,
         Property.NodeScope
     );
 
@@ -250,13 +250,7 @@ public class XPackSettings {
 
     public static final Setting<String> PASSWORD_HASHING_ALGORITHM = defaultStoredPasswordHashAlgorithmSetting(
         "xpack.security.authc.password_hashing.algorithm",
-        (s) -> {
-            if (XPackSettings.FIPS_MODE_ENABLED.get(s)) {
-                return Hasher.PBKDF2_STRETCH.name();
-            } else {
-                return Hasher.BCRYPT.name();
-            }
-        }
+        (s) -> Hasher.PBKDF2_STRETCH.name()
     );
 
     public static final Setting<String> SERVICE_TOKEN_HASHING_ALGORITHM = defaultStoredPasswordHashAlgorithmSetting(
-- 
2.48.1

